glue/viewers/table/qt/tests/test_data_viewer.py

```diff
@@ -460,6 +460,49 @@ def test_incompatible_subset():
     assert refresh2.call_count == 0
 
 
+def test_table_incompatible_attribute():
+    """
+    Regression test for a bug where the table viewer generates an
+    uncaught IncompatibleAttribute error if one toggles the
+    visibility of a dataset iff an invalid subset exists. This
```

Comment:
```suggestion
    visibility of a dataset if an invalid subset exists. This
```
spelling nitpick
And I do not quite understand the comment about the "uncaught" error, since that code is actually in the case intercepting said `IncompatibleAttribute` error.

Reply:
Yeah, it's a bit more subtle than that. The problem is that `_update_visible()` tries to get a mask for all visible layer_artists if (and only if) the underlying dataset is not visible. I have updated the test description to expand on this point.